### PR TITLE
fix: LEAP-1583: Fix comment overlay displaying conditions

### PR DESCRIPTION
**Pull request by Gondragos**
Add checks to ensure comments are shown only if the interfaces allow comments, and if the comment store is commentable.

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
Without that, we can get an error message from the server, as an app may request a list of comments without the right parameters (annotation id), as they can not exist for predictions.

#### What does this fix?
It adds missing conditions not to do anything with comments.

#### What feature flags were used to cover this change?
`fflag_feat_all_leap_1430_per_field_comments_100924_short`

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
`Comments Overlay`

### CODE REVIEW

**Review by hlomzik — APPROVED** [review]

File: web/libs/editor/src/components/App/App.jsx [hunk 1]
```diff
@@ -203,7 +203,10 @@ class App extends Component {
   }
 
   renderCommentsOverlay(selectedAnnotation) {
-    const commentStore = this.props.store.commentStore;
+    const { store } = this.props;
+    const { commentStore } = this.props.store;
```
Comment (yyassi-heartex) [inline web/libs/editor/src/components/App/App.jsx]:
```suggestion
    const { store } = this.props;
    const { commentStore } = store;
```

File: web/libs/editor/src/components/App/App.jsx [hunk 2]
```diff
@@ -203,7 +203,10 @@ class App extends Component {
...
+
+    if (!store.hasInterface("annotations:comments") || !store.commentStore.isCommentable) return null;
```
Comment (yyassi-heartex) [inline web/libs/editor/src/components/App/App.jsx]:
```suggestion
    if (!store.hasInterface("annotations:comments") || !commentStore.isCommentable) return null;
```

**Review by yyassi-heartex — APPROVED** [review]
minor replacement suggestions to leverage the defined variables

**Review by bmartel — APPROVED** [review]

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/11328574618)
> Successfully merged:  create mode 100644 web/libs/editor/src/utils/commentClassification.ts

**Pull request merged**